test/resources/openshift_auth.go

```diff
@@ -0,0 +1,202 @@
+	}
+
+	return openshiftToken, nil
+}
+
+// auth user returning openshift token
+func getOpenshiftToken(client *http.Client, user User, authUrl string) (string, error) {
+	// create query string
+	u, err := query.Values(user)
+	if err != nil {
+		return "", fmt.Errorf("error occured while parsing values, %w", err)
+	}
+
+	// create post to openshift auth
+	req, err := http.NewRequest("POST", authUrl, strings.NewReader(u.Encode()))
+	if err != nil {
+		return "", fmt.Errorf("error occured during new request: %w", err)
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
+	// handle request
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("error occured during request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	// check response for token
+	for _, c := range resp.Cookies() {
+		if c.Name == OpenshiftTokenName {
+			return c.Value, nil
+		}
+	}
+	return "", fmt.Errorf("no openshift session token found")
+}
+
+// state is needed throughout the auth process, this call returns state
+func getOpenshiftState(client *http.Client, stateUrl string) (string, error) {
+	// create new request
+	req, err := http.NewRequest("GET", stateUrl, nil)
+	if err != nil {
+		return "", fmt.Errorf("error occured creating request: %w", err)
+	}
+
+	// handle request
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("error occured at client do: %w", err)
+	}
+
+	// parse url query
+	parsedQuery, err := url.ParseQuery(resp.Request.URL.RawQuery)
+	if err != nil {
+		return "", fmt.Errorf("error occured parsing query: %w", err)
+	}
+	// return state query
+	state := parsedQuery["state"][0]
+	if state == "" {
+		return "", errors.New("failed to find state during parse")
+	}
+	return state, nil
+}
+
+// a session is needed throughout the auth process, this call parses the openshift idp login page to return the correct url with the session query string
+func getOpenshiftAuthUrl(client *http.Client, oauthUrl string, masterUrl string, idp string, state string) (string, error) {
+	// create login query options
+	loginOpt := LoginOptions{"console", idp}
+	lv, _ := query.Values(loginOpt)
+
+	// create callback options
+	callbackOpt := CallbackOptions{"code", "user:full", state}
+	cv, _ := query.Values(callbackOpt)
+
+	// create callback url
+	callbackUrl := "https%3A%2F%2F" + masterUrl + "%2Fauth%2Fcallback&" + cv.Encode()
+	// create request url, not this url needed to be created in this format to avoid any unwanted escape character
+	reqUrl := "https://" + oauthUrl + "/oauth/authorize?" + lv.Encode() + "&redirect_uri=" + callbackUrl
+	req, err := http.NewRequest("GET", reqUrl, nil)
+	if err != nil {
+		return "", fmt.Errorf("error occured while creating a new request, %w", err)
+	}
+
+	// perform request, handling redirects
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("error occured at client do: %w", err)
+	}
+
+	// parse response body
+	doc, err := html.Parse(resp.Body)
```

Comment:
Instead of using the html parser we can introduce the **goquery**, it will simplify the parsing of the html, I'm already using it successfully to authenticate AMQ 
https://github.com/integr8ly/integreatly-operator/blob/1abb8e717b6e19a69e2616d9b6cd025dc362379c/test/common/amq_address_and_address_space.go#L53